src/Native/configure.cmake

```diff
@@ -175,13 +181,27 @@ check_function_exists(
     kqueue
     HAVE_KQUEUE)
 
-check_function_exists(
-    gethostbyname_r
-    HAVE_GETHOSTBYNAME_R)
+check_prototype_definition(
+     gethostbyaddr_r
+     "int gethostbyaddr_r(const void* addr, const socklen_t len, int type, struct hostent* result, char* buffer, size_t buflen, stuct hostent** entry, int* error)"
+     0
+     "sys/types.h;netdb.h"
```

Comment:
Fixed by using check_cxx_source_compile macro. For some reason cmake's check_prototype_definition is using another macro ` cmakeRequireSymbol` which fails but the symbol exists in the file it was looking in.